lib/elixir/test/elixir/kernel_test.exs

```diff
@@ -584,6 +588,11 @@ defmodule KernelTest do
 
       assert put_in(users["john"].age, 28) == %{"john" => %{age: 28}, "meg" => %{age: 23}}
 
+      struct = %StructAccess{foo: %StructAccess{}}
+
+      assert put_in(struct.foo.bar, :baz) ==
+               %StructAccess{bar: nil, foo: %StructAccess{bar: :baz, foo: nil}}
```

Comment:
I added this to the tests of each individual function, if this is more considered an axillary concern I can see how putting all the three examples into separate tests also makes total sense.